cpp/core/shuffle/RoundRobinPartitioner.cc

```diff
@@ -22,15 +22,27 @@ namespace gluten {
 arrow::Status gluten::RoundRobinPartitioner::compute(
     const int32_t* pidArr,
     const int64_t numRows,
-    std::vector<uint16_t>& partitionId,
-    std::vector<uint32_t>& partitionIdCnt) {
-  std::fill(std::begin(partitionIdCnt), std::end(partitionIdCnt), 0);
-  partitionId.resize(numRows);
-  for (auto& pid : partitionId) {
-    pid = pidSelection_;
-    partitionIdCnt[pidSelection_]++;
-    pidSelection_ = (pidSelection_ + 1) == numPartitions_ ? 0 : (pidSelection_ + 1);
+    std::vector<uint16_t>& row2Partition,
+    std::vector<uint32_t>& partition2RowCount) {
+  std::fill(std::begin(partition2RowCount), std::end(partition2RowCount), 0);
+  row2Partition.resize(numRows);
+
+  int32_t pidSelection = 0;
```

Comment:
We should use `pidSelection_` here.
You can use a local variable `int pidSelection` initialized from `pidSelection_` in the following loop, and store it into `pidSelection_` after the loop

Reply:
ok